sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -29,6 +29,11 @@ def test_fit_transform():
     assert_array_equal(expected, est.transform(X))
 
 
+def test_valid_n_bins():
```

Comment:
I might prefer to remove the test
(1) KBinsDiscretizer is now a PR(not in master), so it can be considered an improvement inside the PR.
(2) We do not add too much test in #10017. I don't think it deserves a test.
(3) It seems not a regression test?
```python
isinstance(2, np.int) = True
isinstance(np.array([2])[0], np.int) = True
```

Reply:
(1) This is a PR to the `discrete` branch, which will end up in the PR from `discrete` to `master`.
(2) I would be in favor of this test, even though we may not need something systematic in #10017.
(3) For me:
```py
np.__version__ == '1.13.1'
isinstance(np.array([2])[0], np.int) == False
```

Reply:
My previous script is under python 2.7.12 and numpy 1.13.3 and I can reproduce yours under python 3.5.4 and numpy 1.13.1. So I think the reason for our difference is numpy version.
Let's keep the test if you think it is appropriate :) LGTM for the great example.